ironfish/src/rpc/routes/wallet/getAccountsStatus.ts

```diff
@@ -45,22 +46,26 @@ routes.register<typeof GetAccountStatusRequestSchema, GetAccountStatusResponse>(
   async (request, { node }): Promise<void> => {
     Assert.isNotUndefined(node)
 
-    const headHashes = new Map<string, Buffer | null>()
+    const heads = new Map<string, HeadValue | null>()
     for await (const { accountId, head } of node.wallet.walletDb.loadHeads()) {
-      headHashes.set(accountId, head?.hash ?? null)
+      heads.set(accountId, head)
     }
+
     const accountsInfo: GetAccountStatusResponse['accounts'] = []
     for (const account of node.wallet.listAccounts()) {
-      const headHash = headHashes.get(account.id)
-      const blockHeader = headHash ? await node.chain.getHeader(headHash) : null
-      const headInChain = !!blockHeader
-      const headSequence = blockHeader?.sequence || 'NULL'
+      const head = heads.get(account.id)
+      const blockResponse = head?.hash
+        ? await node.wallet.nodeClient.chain.getBlock({
+            hash: head.hash.toString('hex'),
+          })
+        : null
```

Comment:
Could this reuse the `chainHasBlock` method that we added to the wallet?

Reply:
Yea good call